browser/js/common/factories/SongsFactory.js

```diff
@@ -7,6 +7,19 @@ app.factory('SongsFactory', function ($http) {
       });
     },
 
+    createOrUpdateSong: function (song) {
+      if(song.id) {
+        return $http.get('api/v1/songs/' + song.id)
+          .then(function (response) {
+            let foundSong = response.data;
+              return $http.put('api/v1/songs/' + foundSong.id, song);
+          });
+        } else {
+          delete song.id;
```

Comment:
I think we refactored so that we wouldn't need to use it and forgot to take it out. It works without it so I deleted it.